### ISSUE TITLE
default belongsTo User doesn't work on page refresh caused by validation

### ISSUE THREAD

**Issue by bordieris** [inspheric/nova-defaultable-fields#11]
I have this snippet:
```
    public function fields(Request $request)
    {
        $user = $request->user();
        if(!$user) $user = Auth::user();
        return [
            ID::make()->sortable(),
            Text::make('Code', 'uid')->rules('required')->sortable(),
            BelongsTo::make('Author', 'creator', User::class)->default($user)->display('name'),
        ];
    }
```
When I create a new resource, it works presenting the Author field with correct value, but if I try to save (without value on the Code field, which is required), the Author field is reloaded without value.

**Comment by kitbs** [comment]
Hi @bordieris 

Thanks very much for providing a snippet. Can I ask what version of Nova you are using?

I have tested this today in a new install of v2.1.0 and cannot replicate your issue. I originally built and tested the package in v2.0.3 and did not have this issue then either.

When I click "Create", the page is not refreshed and the Author field is not reloaded, only the validation message is displayed below the Code field.

**When page is first loaded:**
![image](https://user-images.githubusercontent.com/4569320/63431292-730bf980-c416-11e9-8cd3-21109b779565.png)

**After clicking "Create":**
![image](https://user-images.githubusercontent.com/4569320/63431311-7bfccb00-c416-11e9-954c-0076d399a24a.png)

**Comment by bordieris** [comment]
Yes, you're right, the page is not "reloaded" after validation, which works in xhr, but Author field default value still vanishes...
I better investigated since I had other fields in this resource, I tried to isolate only the two fields as in my example, but issue is still there.
In the related model I have as usual:
```
    public function creator(){
        return $this->belongsTo(User::class, 'creator_id', 'id');
    }

```

Nova version 2.1.0

**Comment by kitbs** [comment]
Thanks for the extra information. The relationship looks fine and is exactly how I set my code up when I tried to reproduce your scenario in v2.1.0.

Can you try something else for me to help isolate the problem?

* Remove the `->default($user)` from your Author field (for now)
* Go to a new "Create resource" page
* Choose some value manually in the Author field, and leave the Code field blank
* Click "Create"

Can you tell me whether the Author field still keeps the value you manually selected or if this same issue occurs where the field is cleared after validation fails?

Have you tried using default on a Text field instead of a BelongsTo? Can you tell me if the same issue occurs there?

**Comment by bordieris** [comment]
I tried to remove the `->default($user)` from Author field and the issue is still there, then I guess it's not related to your package, sorry for the wrong report.

By the way I already tried default value with other kind of Fields like text, textarea, boolean and this works as expected also in case of validation errors.

I tried also other belongsTo and it seems not working unless I set them searcheable. In the example we started from, if I change the field as:
  `BelongsTo::make('Author', 'creator', User::class)->default($user)->display('name')->searchable(),`
it works (and also shows a non requested Gravatar)!
